### PR TITLE
Capybra

### PR DESCRIPTION
**Pull request by danielsafari143**
I have added integration tests for my blog app in this pull request, and I have also fixed a performance issue caused by N+1 queries.
change1
change2

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 class PostsController < ApplicationController
   def index
-    @post = User.find_by(id: params['user_id'])
+    @post = User.includes(:posts).find_by(id: params['user_id'])
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [x] @danielsafari143 and @BryteLitty Great work trying to fix the N + 1 problem, however, you did it wrongly, to fix the issue we are supposed to fetch all user's posts and comments at once ge using a code like `@posts = @user.posts.includes(:comments)` k.kindly consider fixing the issue and update the screenshots as well.

File: Gemfile [hunk 2]
```diff
@@ -66,6 +66,7 @@ group :development do
 
   # Speed up commands on slow machines / big apps [https://github.com/rails/spring]
   # gem "spring"
+  gem 'bullet'
```
Comment (danmainah) [inline Gemfile]: - [x] @Team I can see you have gem bullet installed in your app for this milestone, The project requires us not to use this gem but instead use this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
image

File: spec/features/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  it 'has a profile picture' do
+    visit "/users/#{fake_user.id}"
+
+    users = page.all('.profile-picture')
+    users.each do |profile_picture|
+      expect(profile_picture).to have_css('img')
+    end
+  end
+
```
Comment (danmainah) [inline spec/features/show_spec.rb]: - [ ] @Team great work adding tests for the user show page however you forgot to add a test that asserts you can see the user's username.
- [ ] Kindly consider changing the file and description name for all integrated tests files, currently, it's difficult to know which file tests which page.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @danielsafari143 and @BryteLitty , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Used Capybara to write integration tests for each view in the project.
✔️  Added a Professionally written README file with all sections required


# Required Changes ♻️

- [ ] @Team kindly ensure all tests are passing currently some tests are failing.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by prolajumokeoni — APPROVED** [review]
Hi @danielsafari143 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Bullet gem removed ✅ 
- All tests are passing 🚀  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**